platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/MarkerFileWriter.java

```diff
@@ -107,7 +107,16 @@ public void writeMarkerFile(@NonNull final String filename) {
         markerFilesWritten.computeIfAbsent(filename, key -> {
             final Path markerFile = markerFileDirectory.resolve(filename);
             try {
-                Files.createFile(markerFile);
+                // Prevent any other writer for the same string to check and write the file at the same time.
+                // The cost of acquiring the lock is no worse than the file system calls.
+                synchronized (filename) {
```

Comment:
This makes me a bit uneasy. If we are using shared static constants to define the file name strings then I think it will prevent parallel writing... but this is something that seems fragile and is not part of the documented API for this method.

A possible alternate approach might be to accept the fact that there is a race condition, and to log an error in the logs if we fail to write the marker file due to the race condition.

Reply:
If there is concern with tests writing marker files in parallel, I think we should mark these unit tests as `@TimingSensisitive`.